### PR TITLE
Enum methods implementation

### PR DESCRIPTION
**Pull request by yochiwarez**

### CODE REVIEW

File: .cubocop.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+AllCops:
```
Comment (bolah2009) [inline .cubocop.yml]: This filename is `.rubocop.yml` not `.cubocop.yml`

File: enum_methods.rb [hunk 2]
```diff
@@ -1,7 +1,6 @@
 # frozen_string_literal: true
 
 module Enumerable
-
   def my_each
```
Comment (samgaco) [inline enum_methods.rb]: my_each returns an Enumerator if no block is given

File: enum_methods.rb [hunk 3]
```diff
@@ -11,7 +10,7 @@ def my_each
   end
 
   def my_each_with_index
```
Comment (samgaco) [inline enum_methods.rb]: my_each_with_index returns an enumerator if no block is given

File: enum_methods.rb [hunk 4]
```diff
@@ -152,36 +174,103 @@ def my_inject(df = 1,sbl = nil)
...
+puts "my_each_with_index"
+arr = ["Billy", "Wild Bill", "Big Bill", "Coffdrop"]
+arr.my_each_with_index do |nickname, val|
+    puts "String: #{nickname}, Index: #{val}"
+    #prints out each element with index
+end
 
-arr = [2,4,5]
+puts "---------"
+puts "my_select"
```
Comment (samgaco) [inline enum_methods.rb]: my_select returns an enumerator if no block is given

File: enum_methods.rb [hunk 5]
```diff
@@ -22,21 +21,21 @@ def my_select
...
+        if yield(self[i])
           r.push(self[i])
         end
       end
       r
     end
   end
 
-  def my_all?
+  def my_all?(arg1 = nil)
```
Comment (samgaco) [inline enum_methods.rb]: - [ ] my_all? when no block or argument is given returns true when none of the collection members are false or nil
- [ ] my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex
- [ ] my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern

File: enum_methods.rb [hunk 6]
```diff
@@ -152,36 +174,103 @@ def my_inject(df = 1,sbl = nil)
...
 end
 
-# puts g
+puts r
 
-# g = arr.my_inject(2) do |acl, i|
-#   acl - i
-# end
+puts "---------"
+puts "my_any"
```
Comment (samgaco) [inline enum_methods.rb]: - [ ] my_any? when no block or argument is given returns true if at least one of the collection is not false or nil
- [ ] my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class
- [ ] my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex
- [ ] my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern

File: enum_methods.rb [hunk 7]
```diff
@@ -152,36 +174,103 @@ def my_inject(df = 1,sbl = nil)
...
+r = arr.my_any? do |name|
+    name.match(/Bi/) #returns true for the given set
+end
 
-# puts g
+puts r
 
-# g = arr.my_inject(1, :-)
+puts "---------"
+puts "my_none"
```
Comment (samgaco) [inline enum_methods.rb]: - [ ] my_none? when no block or argument is given returns true only if none of the collection members is true
- [ ] my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
- [ ] my_none? when a Regex is passed as an argument returns true only if none of the collection matches the Regex
- [ ] my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern

File: enum_methods.rb [hunk 8]
```diff
@@ -115,32 +140,29 @@ def my_map
     end
   end
 
-  def my_inject(df = 1,sbl = nil)
-
-    
+  def my_inject(arg1 = 1, sbl = nil)
```
Comment (samgaco) [inline enum_methods.rb]: - [ ] my_inject combines all elements of enum by applying a binary operation, specified by a block
- [ ] my_inject when a symbol is specified combines each element of the collection by applying the symbol as a named method

File: enum_methods.rb [hunk 9]
```diff
@@ -152,36 +174,103 @@ def my_inject(df = 1,sbl = nil)
...
+    val < 20 #returns 8 for the given set
+end
+
+run2 = arr.my_count(3) #returns 2 for the given set
+
+run3 = arr.my_count #returns 10 for the given set
+puts "Block: #{run1}, Param: #{run2}, No arg or param: #{run3}"
+
+puts "---------"
+puts "my_map"
```
Comment (samgaco) [inline enum_methods.rb]: - [ ] my_map returns a new array with the results of running block once for every element in enum.
- [ ] my_map returns an Enumerator if no block is given

File: enum_methods.rb (hunk 2 above)
Comment (yochiwarez) [reply enum_methods.rb]: fixed

File: enum_methods.rb (hunk 3 above)
Comment (yochiwarez) [reply enum_methods.rb]: fixed

File: enum_methods.rb (hunk 4 above)
Comment (yochiwarez) [reply enum_methods.rb]: fixed

File: enum_methods.rb (hunk 5 above)
Comment (yochiwarez) [reply enum_methods.rb]: fixed..i think...i hope .. just let me know if not...

File: enum_methods.rb (hunk 6 above)
Comment (yochiwarez) [reply enum_methods.rb]: fixed..i think...i hope .. just let me know if not...

File: enum_methods.rb (hunk 2 above)
Comment (samgaco) [reply enum_methods.rb]: :100:

File: enum_methods.rb (hunk 3 above)
Comment (samgaco) [reply enum_methods.rb]: :100:

File: enum_methods.rb (hunk 4 above)
Comment (samgaco) [reply enum_methods.rb]: :100:

File: enum_methods.rb (hunk 5 above)
Comment (samgaco) [reply enum_methods.rb]: - [ ] my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern

That one still not working.

File: enum_methods.rb (hunk 6 above)
Comment (samgaco) [reply enum_methods.rb]: - [ ] my_any? when no block or argument is given returns true if at least one of the collection is not false or nil
- [ ]  my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern

those are still left to be adjusted

File: enum_methods.rb (hunk 5 above)
Comment (samgaco) [reply enum_methods.rb]:
```
irb(main):250:0> [1,1].all?(1)   
=> true
irb(main):251:0> [1,1].my_all?(1)   
...
TypeError (compared with non class/module)



```

File: enum_methods.rb (hunk 6 above)
Comment (samgaco) [reply enum_methods.rb]:
```
irb(main):254:0> [true, true, false].any?
=> true
irb(main):255:0> [true, true, false].my_any?
=> true
irb(main):256:0> [ false].any?
=> false
irb(main):257:0> [ false].my_any?
=> true
```

Comment (samgaco) [reply enum_methods.rb]:
```
irb(main):258:0> [1,2,99].any?(99)
=> true
irb(main):259:0> [1,2,99].my_any?(99)
....
TypeError (compared with non class/module)


```

File: enum_methods.rb (hunk 5 above)
Comment (yochiwarez) [reply enum_methods.rb]: it is working now, i didn't push my changes yet XD

Comment (yochiwarez) [reply enum_methods.rb]: now this one is working i every cases, when arg1 is given

puts "my_all? regexp #{["1", "9"].my_all?(/\d/)}" => true
puts "my_all? case 1 #{["1", "a"].my_all?(/\d/)}" => false
puts "my_all? case 2 #{["1", "9"].my_all?(String)}" => true
puts "my_all? case 3 #{[1, 2, 3].my_all?(2)}" => false
puts "my_all? case 4 #{[1, 1, 1].my_all?(1)}" => true
puts "my_all? case 5 #{["1", "1"].my_all?("1")}" => true
puts "my_all? case 6 #{[1, 2i, 3.14].all?(Numeric)}" => true

File: enum_methods.rb (hunk 6 above)
Comment (yochiwarez) [reply enum_methods.rb]: now working on all this cases

puts "my_any? regexp #{["1", "9"].my_any?(/\d/)}"
puts "my_any? case 1 #{["b", "a"].my_any?(/\d/)}"
puts "my_any? case 2 #{["1", 1].my_any?(String)}"
puts "my_any? case 3 #{[1, 2, 3].my_any?(2)}"
puts "my_any? case 4 #{[3, 4, 8].my_any?(1)}"
puts "my_any? case 5 #{["1", "1"].my_any?("1")}"
puts "my_any? case 6 #{[1, 2i, "nomueric"].my_any?(Numeric)}"
puts "my_any? case 7 #{[nil, false , "gg", 48].my_any?}"
puts "my_any? case 8 #{[true, true, true].my_any?}"
puts "my_any? case 9 #{[true, true, "f", 1].my_any?}"

my_any? regexp true
my_any? case 1 false
my_any? case 2 true
my_any? case 3 true
my_any? case 4 false
my_any? case 5 true
my_any? case 6 true
my_any? case 7 true
my_any? case 8 true
my_any? case 9 true

File: enum_methods.rb (hunk 7 above)
Comment (yochiwarez) [reply enum_methods.rb]: puts "my_none? regexp #{["1", "9"].my_none?(/\d/)}"
puts "my_none? case 1 #{["b", "a"].my_none?(/\d/)}"
puts "my_none? case 2 #{[4, 1].my_none?(String)}"
puts "my_none? case 3 #{[1, 2, 3].my_none?(2)}"
puts "my_none? case 4 #{[3, 4, 8].my_none?(1)}"
puts "my_none? case 5 #{["1", "1", 4].my_none?("1")}"
puts "my_none? case 6 #{[1, 2i, "nomueric"].my_none?(Numeric)}"
puts "my_none? case 7 #{[nil, false, "gg", 48].my_none?}"
puts "my_none? case 8 #{[false, false, nil].my_none?}"
puts "my_none? case 9 #{[true, true, "f", 1].my_none?}\n\n\n"

my_none? regexp false
my_none? case 1 true
my_none? case 2 true
my_none? case 3 false
my_none? case 4 true
my_none? case 5 false
my_none? case 6 false
my_none? case 7 false
my_none? case 8 true
my_none? case 9 false

File: enum_methods.rb (hunk 8 above)
Comment (yochiwarez) [reply enum_methods.rb]: tested with this cases
puts "#{(5..10).my_inject { |sum, n| sum + n }}"
puts "#{[5, 6, 7, 8, 9, 10].my_inject { |sum, n| sum + n }}"
puts "#{(5..10).my_inject(:+)}"
puts "#{(5..10).my_inject(1) { |product, n| product * n }}"
puts "#{[5, 6, 7, 8, 9, 10].my_inject(1) { |product, n| product * n }}"
puts "#{(5..10).my_inject(2) { |sum, n| sum + n }}"
puts "#{[5, 6, 7, 8, 9, 10].my_inject(3) { |sum, n| sum + n }}"
puts "#{(5..10).my_inject(:+)}"
puts "#{(5..10).my_inject(3) { |product, n| product * n }}"
puts "#{[5, 6, 7, 8, 9, 10].my_inject(2) { |product, n| product * n }}\n\n\n"

45
45
45
151200
151200
47
48
45
453600
302400

File: enum_methods.rb (hunk 9 above)
Comment (yochiwarez) [reply enum_methods.rb]: working on these cases 
puts "#{(1..4).my_map { |i| i * i }}"
puts "#{(1..4).my_map.to_a}"
puts "#{[1, 2, 3, 4].my_map { |i| i * i }}"
puts "#{[1, 2, 3, 4].my_map.to_a}\n\n\n"

[1, 4, 9, 16]
[1, 2, 3, 4]
[1, 4, 9, 16]
[1, 2, 3, 4]

File: .rubocop.yml [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+  Enabled: false
+
+Metrics/CyclomaticComplexity:
+  Enabled: false
+
+Metrics/PerceivedComplexity:
+  Enabled: false
+
+Style:
+  Enabled: false
```
Comment (Torres-ssf) [inline .rubocop.yml]: **[REQUIRED]**
We need to remove these lines, they do not belong to the [original](https://github.com/microverseinc/linters-config/blob/master/ruby/.rubocop.yml) .rubocop.yml created by Microverse.

File: .rubocop.yml [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+
```
Comment (Torres-ssf) [inline .rubocop.yml]: **[REQUIRED]**
We need to remove these lines, they do not belong to the [original](https://github.com/microverseinc/linters-config/blob/master/ruby/.rubocop.yml) .rubocop.yml created by Microverse.

Comment (yochiwarez) [reply .rubocop.yml]: i will remove them

**Pull request closed without merging**